lib/discuss/auth/process.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Discuss.Auth.Process do
+  alias Discuss.Auth
+
+  @spec perform(map) :: {:ok, Auth.User} | {:error, User.changeset()}
+  @doc """
+  Check if a user is already into the database with given email. If yes, update token and provider.
+  If not, create a new user.
+
+  ## Examples
+
+      iex> perform(%{email: "john@email.com", provider: "github", token: "123456789"})
+      {:ok, %Auth.User{}}
+
+      iex> perform(%{email: "john@email.com", provider: "invalid", token: nil})
+      {:error, %Auth.User.changeset{}}
+  """
+  def perform(params) when is_map(params) and is_map_key(params, :email) do
```

Comment:
Eu chamaria essa função de `upsert` e a colocaria no context de user

Reply:
Tb não vejo necessidade de usar guard_clause

Reply:
seria upset? `upsert` ta dando como erro de digitação aqui no linter rsrs

Reply:
upsert = update/insert

https://hexdocs.pm/ecto/constraints-and-upserts.html#content